compiler/rustc_resolve/src/imports.rs

```diff
@@ -1145,7 +1168,7 @@ impl<'a, 'b> ImportResolver<'a, 'b> {
         }
 
         // Record the destination of this import
-        self.r.record_partial_res(import.id, PartialRes::new(module.res().unwrap()));
+        self.r.record_partial_res(import.expect_id(), PartialRes::new(module.res().unwrap()));
```

Comment:
This one is a bit more obvious, though could you add a assert stating that the method must only be called for `ImportKind::Glob`?